LeaderboardBackend/Controllers/UsersController.cs

```diff
@@ -51,45 +53,50 @@ public async Task<ActionResult<UserViewModel>> GetUserById(Guid id)
     /// </param>
     /// <response code="201">The `User` was registered and returned successfully.</response>
     /// <response code="400">
-    ///     The passwords did not match or the request was otherwise malformed.
+    ///     The request was malformed.
+    /// </response>
+    /// <response code="422">
+    ///     The request contains errors.<br/><br/>
+    ///     Validation error codes by property:
+    ///     - **Username**:
+    ///       - **UsernameFormat**: Invalid username format
+    ///     - **Password**:
+    ///       - **PasswordFormat**: Invalid password format
+    ///     - **Email**:
+    ///       - **EmailValidator**: Invalid email format
     /// </response>
     /// <response code="409">
-    ///     A `User` with the specified username or email already exists.
+    ///     A `User` with the specified username or email already exists.<br/><br/>
+    ///     Validation error codes by property:
+    ///     - **Username**:
+    ///       - **UsernameTaken**: the username is already in use
+    ///     - **Email**:
+    ///       - **EmailAlreadyUsed**: the email is already in use
     /// </response>
     [AllowAnonymous]
     [HttpPost("register")]
+    [ApiConventionMethod(typeof(Conventions), nameof(Conventions.PostAnon))]
     [ProducesResponseType(StatusCodes.Status201Created)]
-    [ProducesResponseType(StatusCodes.Status400BadRequest)]
```

Comment:
Can't this still produce a 400 due to, say, malformed JSON?

Reply:
This is now included in the ApiConvention (line 78)
400 is still documented:
![image](https://github.com/leaderboardsgg/leaderboard-backend/assets/1781420/ad0839d1-2c69-4afe-885d-9a7cb7f9de86)